pkg/datasource/handler.go

```diff
@@ -247,7 +247,7 @@ func (s *Service) downloadFromURL(parameters map[string]string) error {
 	}
 
 	go func() {
-		_, err := s.downloader.DownloadFile(s.ctx, url, s.tmpFilePath, s)
+		_, err := s.downloader.DownloadFile(s.ctx, url, s.tmpFilePath, s.size, s)
```

Comment:
So determining the need to process a possible sparse download file is based on if the size is able to know first? Why not just always do skip zero bytes from the download file?

Reply:
OK, I just realized the downloader can calculate the file size after the download then retries the size truncate anyway.
In the current design, the truncate happens before the downloading. If the size is unknown and the truncate is skipped, the zero-byte skipping would lead to the mismatching between the final file size and the expected file size, which then causes the checksum mismatching. But if we can retry truncate after the downloading, this mismatching may not happen.

Reply:
I think we are fine to keep the current implementation or do you want to enhance it a bit for truncating after the download?

Reply:
Let me doubt-check it later. I prefer to improve it if possible. Actually, I don't know if there is a URL that would fail the size detection but allow the downloading...